Change the soc from ace30_ptl to ace30

gbernatxintel opened the pull request:
Renamed soc from ace30_ptl to ace30.
We were previously using the wrong soc name.
The correct name is ace30.

There is only one ptl platform, but there can be several ace30 platforms.

Check the commit on the sof project repo:
Change the soc from ace30_ptl to ace30

Dependencies:
https://github.com/thesofproject/sof/pull/9475

nordicjm commented on soc/intel/intel_adsp/ace/Kconfig.soc:
```diff
@@ -32,9 +32,9 @@ config SOC_SERIES
 	default "ace15_mtpm" if SOC_INTEL_ACE15_MTPM
 	default "ace20_lnl" if SOC_INTEL_ACE20_LNL
-	default "ace30_ptl" if SOC_INTEL_ACE30_PTL
+	default "ace30" if SOC_INTEL_ACE30
```
these should match `SOC_*` should be the value of * in the string

gbernatxintel replied:
In my opinion it is ok, if you see other platforms the names are entered without soc_* just simply string. For other manufacturers I randomly checked and also they don't all use soc_* names but just the name of the specific chip

nordicjm replied:
I mean this should either be `default "intel_ace30" if SOC_INTEL_ACE30` or `default "ace30" if SOC_ACE30` if not then this will fail at some point in the future when the build system generates these symbols automatically from the soc.yml files

gbernatxintel replied:
Oki, I understand, but I still don't think it's a problem. The SOC is selected when we have a given board declared. I think it should be added as it is, and note that other intel and non intel platforms also give the name alone, without the company name prefix

gbernatxintel replied:
Oki, I changed to  ```"ace30" if SOC_ACE30```

nordicjm commented on boards/intel/adsp/Kconfig.intel_adsp:
```diff
@@ -8,5 +8,5 @@ config BOARD_INTEL_ADSP
 	select SOC_INTEL_ACE15_MTPM if BOARD_INTEL_ADSP_ACE15_MTPM_SIM
 	select SOC_INTEL_ACE20_LNL if BOARD_INTEL_ADSP_ACE20_LNL
 	select SOC_INTEL_ACE20_LNL if BOARD_INTEL_ADSP_ACE20_LNL_SIM
-	select SOC_INTEL_ACE30_PTL if BOARD_INTEL_ADSP_ACE30_PTL
-	select SOC_INTEL_ACE30_PTL if BOARD_INTEL_ADSP_ACE30_PTL_SIM
+	select SOC_ACE30 if BOARD_INTEL_ADSP_ACE30_PTL
```
Sorry was not clear, did not mean it needed to be done right now in this PR but just a note about fixing it in a later PR. All the socs will need fixing, I don't know the timeframe for when the feature for automatically generating symbols is planned however

nashif commented on boards/intel/adsp/board.yml:
```diff
@@ -11,6 +11,7 @@ boards:
         variants:
-          - name: 'sim'
+          - name: 'ptl'
+          - name: 'ptl_sim'
```
prefer:
```
      - name: ace30
        variants:
          - name: 'ptl'
            variants:
              - name: 'sim'
          ```

gbernatxintel replied:
Done

dcpleung commented on soc/intel/intel_adsp/ace/Kconfig.soc:
```diff
@@ -20,7 +20,7 @@ config SOC_INTEL_ACE20_LNL
 	  ACE 2.0 Lunar Lake PCH
 
-config SOC_INTEL_ACE30_PTL
+config SOC_ACE30
```
Please keep the Intel name here to act as namespace: `SOC_INTEL_ACE30`.

gbernatxintel replied:
Changed to ```SOC_INTEL_ACE30```

dcpleung reviewed (changes requested):
Also, could you separate the whitespace changes into another commit? A simple rename without any changes should show up in git as a simple rename, and makes it easier to review the diff.

dcpleung commented on boards/intel/adsp/intel_adsp_ace30_ptl.yaml:
This needs to be renamed to remove `_ptl` suffix.

dcpleung commented on boards/intel/adsp/intel_adsp_ace30_ptl_sim.yaml:
Why are you removing this file, instead of renaming it?

gbernatxintel replied:
Yes, as far as the ptl_sim.yaml file is concerned I deleted it by accident. It should be as it is, i.e. changed identifier line to ```identifier: intel_adsp/ace30/ptl/sim```

gbernatxintel commented:
> Also, could you separate the whitespace changes into another commit? A simple rename without any changes should show up in git as a simple rename, and makes it easier to review the diff.

In fact, it's hard to carve out a separate commit, because the whole commit is a name change from ace30_ptl to ace30. The general problem was that the soc was misspelled. If there is for example another board based on the same soc then there will be confusion with the definitions. 
There are no functional changes in this commit

dcpleung commented:
> > Also, could you separate the whitespace changes into another commit? A simple rename without any changes should show up in git as a simple rename, and makes it easier to review the diff.
> 
> In fact, it's hard to carve out a separate commit, because the whole commit is a name change from ace30_ptl to ace30. The general problem was that the soc was misspelled. If there is for example another board based on the same soc then there will be confusion with the definitions. There are no functional changes in this commit

When you look at the diff, there are bunch of header files with only whitespace changes and nothing else. There are not `ace30_ptl` to `ace30` changes in there. All I ask is to separate those simple whitespace changes into another commit so that the renaming would be more simple and direct in git log.

dcpleung commented on soc/intel/intel_adsp/ace/Kconfig.soc:
```diff
@@ -32,9 +32,9 @@ config SOC_SERIES
 	default "intel_ace15_mtpm" if SOC_INTEL_ACE15_MTPM
 	default "intel_ace15_mtpm" if SOC_INTEL_ACE20_LNL
-	default "intel_ace30_ptl" if SOC_INTEL_ACE30_PTL
+	default "intel_ace30" if SOC_INTEL_ACE30
```
Also, keep the toolchain name as `intel_ace30_ptl` as it is related to the Zephyr SDK, and the toolchain there is named with `_ptl`. Changing this would mean Zephyr SDK can no longer build this platform.

gbernatxintel committed:
soc: intel: renamed soc from ace30_ptl to ace30

Renamed soc from ace30_ptl to ace30.
We were previously using the wrong soc name.
The correct name is ace30.

There is only one ptl platform, but there can be several ace30 platforms.

Signed-off-by: Grzegorz Bernat <grzegorzx.bernat@intel.com>

gbernatxintel committed:
soc: intel: Fix problems with the formatter

No functional changes were made in this update.
Only code formatting issues were corrected.

This commit is necessary to preserve Git history
continuity for future changes involving the switch from ace30_ptl to ace30.

Signed-off-by: Grzegorz Bernat <grzegorzx.bernat@intel.com>

dcpleung commented on dts/xtensa/intel/intel_adsp_ace30.dtsi:
You need to remove `intel_adsp_ace30_ptl.dtsi`.

nordicjm commented:
This has broken CI https://github.com/zephyrproject-rtos/zephyr/actions/runs/11009734491/job/30570116372 please submit a fix

aescolar commented:
> This has broken CI https://github.com/zephyrproject-rtos/zephyr/actions/runs/11009734491/job/30570116372 please submit a fix

Thanks @nordicjm revert queued in https://github.com/zephyrproject-rtos/zephyr/pull/78930

gbernatxintel commented:
Thank you @nordicjm ,I'm checking what's happening in this CI test
